File: frontend/src/app/models/TaskType.ts
export enum TaskType {
  ATTRIBUTE_MODEL = 0,
  VALUE_MODEL = 1,
  MESSAGE_MAPPINGS = 2,
  TERM_MAPPINGS = 3,
}

export class TaskTypeHelper {
  public static getName(taskType: TaskType): string {
    switch (taskType) {
      case TaskType.ATTRIBUTE_MODEL: return 'Attribute model';
      case TaskType.VALUE_MODEL: return 'Value model';
      case TaskType.MESSAGE_MAPPINGS: return 'Message Mappings';
      case TaskType.TERM_MAPPINGS: return 'Term Mappings';
      default: return 'Unknown';
    }
  }
}

File: common/src/main/java/org/endeavourhealth/im/common/models/TaskType.java
package org.endeavourhealth.im.common.models;

import com.fasterxml.jackson.annotation.JsonValue;

public enum TaskType {
    ATTRIBUTE_MODEL ((byte)0, "Attribute Model"),
    VALUE_MODEL ((byte)1, "Value Model"),
    MESSAGE_MAPPINGS((byte)2, "Message Mappings"),
    TERM_MAPPINGS((byte)3, "Term Mappings");

    private final Byte value;
    private final String name;
    TaskType(byte value, String name) {
        this.value = value;
        this.name = name;
    }

    @JsonValue
    public byte getValue() { return this.value; }
    public String getName() { return this.name; }
    public String toString() {
        return this.value.toString();
    }

    public static TaskType byValue(Byte value) {
        if (value == null)
            return null;

        for (TaskType t: TaskType.values()) {
            if (t.value.equals(value))
                return t;
        }

        return null;
    }
}
